File: app/src/main/java/com/dt/manager/ui/ApkInfoDialog.java
package com.dt.manager.ui;

import android.app.Dialog;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.Window;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.dt.manager.R;
import com.dt.manager.core.ApkInfo;

import java.io.File;

/**
 * Modal dialog showing APK metadata with three actions:
 * FUNCTION, VIEW, INSTALL — matches the MT Manager reference screenshot.
 */
public class ApkInfoDialog {

    public interface Listener {
        void onView(File apkFile);
        void onInstall(File apkFile);
        void onFunctions(File apkFile);
    }

    public static void show(@NonNull AppCompatActivity activity, @NonNull File apkFile, Listener listener) {
        Context ctx = activity.getApplicationContext();
        LayoutInflater inflater = LayoutInflater.from(activity);

        ApkInfo info = ApkInfo.fromFile(ctx, apkFile);

        View content = inflater.inflate(R.layout.dialog_apk_info, null, false);

        Dialog dialog = new Dialog(activity);
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE);
        dialog.setContentView(content);

        ImageView iconApp = content.findViewById(R.id.iconApp);
        TextView textAppName = content.findViewById(R.id.textAppName);
        TextView textVersion = content.findViewById(R.id.textVersion);

        if (info.icon != null) {
            iconApp.setImageDrawable(info.icon);
        } else {
            iconApp.setImageResource(R.drawable.ic_apk);
        }
        textAppName.setText(info.appName != null ? info.appName : apkFile.getName());
        textVersion.setText("Version " + (info.versionName != null ? info.versionName : ""));

        setRow(content, R.id.rowPackage, "Package name", info.packageName);
        setRow(content, R.id.rowVersionCode, "Version code", String.valueOf(info.versionCode));
        setRow(content, R.id.rowFileSize, "File size", info.formatSize());
        setRow(content, R.id.rowSignature, "Signature", info.signatureScheme);
        setRow(content, R.id.rowTargetSdk, "Target SDK", info.formatSdk());
        setRow(content, R.id.rowMinSdk, "Minimum SDK", info.formatMinSdk());

        TextView btnFunctions = content.findViewById(R.id.btnFunctions);
        TextView btnView = content.findViewById(R.id.btnView);
        TextView btnInstall = content.findViewById(R.id.btnInstall);

        btnFunctions.setOnClickListener(v -> {
            if (listener != null) listener.onFunctions(apkFile);
            dialog.dismiss();
        });
        btnView.setOnClickListener(v -> {
            if (listener != null) listener.onView(apkFile);
            dialog.dismiss();
        });
        btnInstall.setOnClickListener(v -> {
            if (listener != null) listener.onInstall(apkFile);
            dialog.dismiss();
        });

        dialog.show();
        // Make dialog fill width
        if (dialog.getWindow() != null) {
            dialog.getWindow().setLayout(
                    (int) (activity.getResources().getDisplayMetrics().widthPixels * 0.92),
                    android.view.ViewGroup.LayoutParams.WRAP_CONTENT);
        }
    }

    private static void setRow(View root, int rowId, String label, String value) {
        View row = root.findViewById(rowId);
        if (row == null) return;
        TextView l = row.findViewById(R.id.label);
        TextView v = row.findViewById(R.id.value);
        if (l != null) l.setText(label);
        if (v != null) v.setText(value != null && !value.isEmpty() ? value : "—");
    }
}

File: app/src/main/java/com/dt/manager/ui/ApkInfoDialog.kt
package com.dt.manager.ui

import android.app.Dialog
import android.view.LayoutInflater
import android.view.View
import android.view.Window
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.dt.manager.R
import com.dt.manager.core.ApkInfo
import java.io.File

/**
 * Modal dialog showing APK metadata with three actions:
 * FUNCTION, VIEW, INSTALL — matches the MT Manager reference screenshot.
 */
object ApkInfoDialog {

    interface Listener {
        fun onView(apkFile: File)
        fun onInstall(apkFile: File)
        fun onFunctions(apkFile: File)
    }

    @JvmStatic
    fun show(activity: AppCompatActivity, apkFile: File, listener: Listener?) {
        val ctx = activity.applicationContext
        val inflater = LayoutInflater.from(activity)

        val info = ApkInfo.fromFile(ctx, apkFile)

        val content = inflater.inflate(R.layout.dialog_apk_info, null, false)

        val dialog = Dialog(activity)
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        dialog.setContentView(content)

        val iconApp: ImageView = content.findViewById(R.id.iconApp)
        val textAppName: TextView = content.findViewById(R.id.textAppName)
        val textVersion: TextView = content.findViewById(R.id.textVersion)

        if (info.icon != null) {
            iconApp.setImageDrawable(info.icon)
        } else {
            iconApp.setImageResource(R.drawable.ic_apk)
        }
        textAppName.text = info.appName ?: apkFile.name
        textVersion.text = "Version " + (info.versionName ?: "")

        setRow(content, R.id.rowPackage, "Package name", info.packageName)
        setRow(content, R.id.rowVersionCode, "Version code", info.versionCode.toString())
        setRow(content, R.id.rowFileSize, "File size", info.formatSize())
        setRow(content, R.id.rowSignature, "Signature", info.signatureScheme)
        setRow(content, R.id.rowTargetSdk, "Target SDK", info.formatSdk())
        setRow(content, R.id.rowMinSdk, "Minimum SDK", info.formatMinSdk())

        val btnFunctions: TextView = content.findViewById(R.id.btnFunctions)
        val btnView: TextView = content.findViewById(R.id.btnView)
        val btnInstall: TextView = content.findViewById(R.id.btnInstall)

        btnFunctions.setOnClickListener {
            listener?.onFunctions(apkFile)
            dialog.dismiss()
        }
        btnView.setOnClickListener {
            listener?.onView(apkFile)
            dialog.dismiss()
        }
        btnInstall.setOnClickListener {
            listener?.onInstall(apkFile)
            dialog.dismiss()
        }

        dialog.show()
        dialog.window?.setLayout(
            (activity.resources.displayMetrics.widthPixels * 0.92).toInt(),
            android.view.ViewGroup.LayoutParams.WRAP_CONTENT
        )
    }

    private fun setRow(root: View, rowId: Int, label: String, value: String?) {
        val row = root.findViewById<View>(rowId) ?: return
        val l: TextView? = row.findViewById(R.id.label)
        val v: TextView? = row.findViewById(R.id.value)
        l?.text = label
        v?.text = if (!value.isNullOrEmpty()) value else "—"
    }
}
